_clients/ruby.md

```diff
@@ -0,0 +1,528 @@
+
+To index one document, use the `index` method:
+
+```ruby
+document = {
+    first_name: 'Connor',
+    last_name: 'James',
+    gpa: 3.93,
+    grad_year: 2021
+}
+  
+client.index(
+    index: 'students',
+    body: document,
+    id: 100,
+    refresh: true
+)
+```
+
+## Updating a document
+
+To update a document, use the `update` method:
+
+```ruby
+client.update(index: 'students', 
+              id: 100, 
+              body: { doc: { gpa: 3.25 } }, 
+              refresh: true)
+```
+
+## Deleting a document
+
+To delete a document, use the `delete` method:
+
+```ruby
+client.delete(
+    index: 'students',
+    id: 100,
+    refresh: true
+)
+```
+
+## Bulk operations
+
+You can perform several operations at the same time by using the `bulk` method. The operations may be of the same type or of different types.
+
+You can index multiple documents as follows:
+
+```ruby
+actions = [
+    { index: { _index: 'students', _id: '200' } },
+    { first_name: 'James', last_name: 'Rodriguez', gpa: 3.91, grad_year: 2019 },
+    { index: { _index: 'students', _id: '300' } },
+    { first_name: 'Nikki', last_name: 'Wolf', gpa: 3.87, grad_year: 2020 }
+]
+client.bulk(body: actions, refresh: true)
+```
+
+You can delete multiple documents as follows:
+
+```ruby
+# Deleting multiple documents.
+actions = [
+    { delete: { _index: 'students', _id: 200 } },
+    { delete: { _index: 'students', _id: 300 } }
+]
+client.bulk(body: actions, refresh: true)
+```
+
+You can perform different operations when using `bulk` as follows:
+
+```ruby
+actions = [
+    { index:  { _index: 'students', _id: 100, data: { first_name: 'Paulo', last_name: 'Santos', gpa: 3.29, grad_year: 2022 } } },
+    { index:  { _index: 'students', _id: 200, data: { first_name: 'Shirley', last_name: 'Rodriguez', gpa: 3.92, grad_year: 2020 } } },
+    { index:  { _index: 'students', _id: 300, data: { first_name: 'Akua', last_name: 'Mansa', gpa: 3.95, grad_year: 2022 } } },
+    { index:  { _index: 'students', _id: 400, data: { first_name: 'John', last_name: 'Stiles', gpa: 3.72, grad_year: 2019 } } },
+    { index:  { _index: 'students', _id: 500, data: { first_name: 'Li', last_name: 'Juan', gpa: 3.94, grad_year: 2022 } } },
+    { index:  { _index: 'students', _id: 600, data: { first_name: 'Richard', last_name: 'Roe', gpa: 3.04, grad_year: 2020 } } },
+    { update: { _index: 'students', _id: 100, data: { doc: { gpa: 3.73 } } } },
+    { delete: { _index: 'students', _id: 200  } }
+]
+client.bulk(body: actions, refresh: true)
+```
+
+In the above example, you pass the data and the header together and you denote the data with the `data:` key.
+
+## Searching for a document
+
+To search for a document, use the `search` method. The following example searches for a student whose first or last name is "James." It uses a `multi_match` query to search for two fields (`first_name` and `last_name`), and it is boosting the `last_name` field in relevance with a caret notation (`last_name^2`). 
```

Comment:
Instead of "it is boosting", can we say "it boosts"? Instead of "with a caret notation", can we say "by using a caret notation"?